jolt-core/src/jolt/instruction/movsign.rs

```diff
@@ -0,0 +1,119 @@
+use rand::prelude::StdRng;
+use rand::RngCore;
+use serde::{Deserialize, Serialize};
+
+use super::JoltInstruction;
+use crate::{
+    jolt::{
+        instruction::SubtableIndices,
+        subtable::{sign_extend::SignExtendSubtable, LassoSubtable},
+    },
+    poly::field::JoltField,
+    utils::instruction_utils::chunk_operand_usize,
+};
+
+#[derive(Copy, Clone, Default, Debug, Serialize, Deserialize)]
+pub struct MOVSIGNInstruction<const WORD_SIZE: usize>(pub u64);
+
+// Constants for 32-bit and 64-bit word sizes
+const NEGATIVE_32: u64 = 0xFFFF_FFFF;
+const NEGATIVE_64: u64 = 0xFFFF_FFFF_FFFF_FFFF;
+const POSITIVE: u64 = 0;
+const SIGN_BIT_32: u64 = 0x8000_0000;
+const SIGN_BIT_64: u64 = 0x8000_0000_0000_0000;
+
+impl<const WORD_SIZE: usize> JoltInstruction for MOVSIGNInstruction<WORD_SIZE> {
+    fn operands(&self) -> (u64, u64) {
+        (self.0, 0)
+    }
+
+    fn combine_lookups<F: JoltField>(&self, vals: &[F], C: usize, _: usize) -> F {
+        let val = vals[0];
+        let mut result = F::zero();
+        for i in 0..C {
```

Comment:
actually could we change this to: 
```suggestion
        for i in 0..WORD_SIZE / 16 {
```